Improve the onboarding access checklist for contributors, maintainers and administrators (#2656)
GenericMappingTools/pygmt: doc/maintenance.md

## Onboarding Access Checklist

### As a Contributor

> 

Reviewer: I would move the "pygmt-contribuors team" point to the top:
```suggestion

- Added to the [pygmt-contributors](https://github.com/orgs/GenericMappingTools/teams/pygmt-contributors) team (gives 'write' permission to the repository)
```

- Added as collaborator on [DAGsHub](https://dagshub.com/GenericMappingTools/pygmt/settings/collaboration) (gives 'write' permission to dvc remote storage)

> - Added to [pygmt-contributors](https://github.com/orgs/GenericMappingTools/teams/pygmt-contributors) team (gives 'write' permission to the repository)

Reviewer: ```suggestion

```

- Added as member on the [PyGMT devs Slack channel](https://pygmtdevs.slack.com) (for casual conversations)
- Added as member on [HackMD](https://hackmd.io/@pygmt) (for draft annoucements) [optional]

### As a Maintainer

> - Added to [pygmt-maintainers](https://github.com/orgs/GenericMappingTools/teams/pygmt-maintainers) team (gives 'maintain' permission to the repository)

Reviewer: ```suggestion
- Added to the [pygmt-maintainers](https://github.com/orgs/GenericMappingTools/teams/pygmt-maintainers) team (gives 'maintain' permission to the repository)
```

- Update the roles in the {doc}`Team Gallery <team>` page
- Added as moderator on [GMT forum](https://forum.generic-mapping-tools.org) (to see mod-only discussions)

### As an Administrator

> 

Reviewer: I would the "pygmt-admin team" point to the top:
```suggestion

- Added to the [pygmt-admin](https://github.com/orgs/GenericMappingTools/teams/pygmt-admin) team (gives 'admin' permission to the repository)
```

- Added to the [pygmt-admin](https://github.com/orgs/GenericMappingTools/teams/pygmt-admin) team (gives 'admin' permission to the repository)

> - Added as maintainer on [PyPI](https://pypi.org/project/pygmt/) and [Test PyPI](https://test.pypi.org/project/pygmt) [optional]

Reviewer 2: Include admin permissons on DAGsHub too? Mark as optional or not?

```suggestion
- Added as admin on [DAGsHub](https://www.dagshub.com/GenericMappingTools/pygmt/settings/collaboration)
- Added as maintainer on [PyPI](https://pypi.org/project/pygmt/) and [Test PyPI](https://test.pypi.org/project/pygmt) [optional]
```

> - Added to [pygmt-admin](https://github.com/orgs/GenericMappingTools/teams/pygmt-admin) (gives 'admin' permission to the repository)

Reviewer: ```suggestion

```

## Onboarding Access Checklist

Note that anyone can contribute to PyGMT, even without being added to the
[GenericMappingTools team](https://github.com/orgs/GenericMappingTools/teams).
The onboarding items below are for people who would like to make regular
contributions, and could benefit from extra permissions to the developer and

> communication tools we use.
> ### As a Contributor

Reviewer 2: ```suggestion
communication tools we use.

### As a Contributor
```

> - Added to the [pygmt-contributors](https://github.com/orgs/GenericMappingTools/teams/pygmt-contributors) team (gives 'write' permission to the repository)

Reviewer: For consistency with line 13, I would include "team" in the link.
```suggestion
- Added to the [pygmt-contributors team](https://github.com/orgs/GenericMappingTools/teams/pygmt-contributors) (gives 'write' permission to the repository)
```

- Added as collaborator on [DAGsHub](https://dagshub.com/GenericMappingTools/pygmt/settings/collaboration) (gives 'write' permission to dvc remote storage)
- Added as member on the [PyGMT devs Slack channel](https://pygmtdevs.slack.com) (for casual conversations)
- Added as member on [HackMD](https://hackmd.io/@pygmt) (for draft announcements) [optional]

### As a Maintainer

> - Added to the [pygmt-maintainers](https://github.com/orgs/GenericMappingTools/teams/pygmt-maintainers) team (gives 'maintain' permission to the repository)

Reviewer: ```suggestion
- Added to the [pygmt-maintainers team](https://github.com/orgs/GenericMappingTools/teams/pygmt-maintainers) (gives 'maintain' permission to the repository)
```

> - Update the roles in the {doc}`Team Gallery <team>` page

Reviewer: Not completly sure here, but maybe singular makes for sense here?

```suggestion
- Update the role on the {doc}`Team Gallery page <team>`
```

- Added as moderator on the [GMT forum](https://forum.generic-mapping-tools.org) (to see mod-only discussions) [optional]
- Added as a maintainer on [Readthedocs](https://readthedocs.org/projects/pygmt-dev) [optional]

### As an Administrator

> - Added to the [pygmt-admin](https://github.com/orgs/GenericMappingTools/teams/pygmt-admin) team (gives 'admin' permission to the repository)

Reviewer: ```suggestion
- Added to the [pygmt-admin team](https://github.com/orgs/GenericMappingTools/teams/pygmt-admin) (gives 'admin' permission to the repository)
```
